kernel/kernel-api/src/main/java/io/delta/kernel/internal/util/SchemaUtils.java

```diff
@@ -303,6 +315,37 @@ private static String escapeDots(String name) {
     return name.contains(".") ? "`" + name + "`" : name;
   }
 
+  private static void collectLeafColumnsInternal(
+      StructType schema,
+      Column parentColumn,
+      Set<String> excludedColumns,
+      List<Column> result,
+      int maxColumns) {
+    for (StructField field : schema.fields()) {
+      if (result.size() >= maxColumns) {
+        return;
+      }
+
+      Column currentColumn = null;
```

Comment:
nit: you can do `final Column currentColumn;` // do can avoid assigning to null and then re-assigning later